SingularityUI/app/views/TaskPoller.coffee

```diff
@@ -0,0 +1,133 @@
+HistoricalTasks = require '../collections/HistoricalTasks'
+
+TaskFiles = require '../collections/TaskFiles'
+RequestTasks = require '../collections/RequestTasks'
+
+taskPollerWaitingTemplate = require 'templates/vex/taskPollerWaiting'
+taskPollingFailureTemplate = require 'templates/vex/taskPollingFailure'
+
+vex = require 'vex.dialog'
+
+interval = (a, b) -> setInterval(b, a)  # f u javascript
+timeout = (a, b) -> setTimeout(b, a)
+
+TIMEOUT_MINUTES = 1 # Modify this
+TIMEOUT_SECONDS = TIMEOUT_MINUTES * 60 # Don't modify this (unless you want a timeout less than a minute)
+TIMEOUT_MILLISECONDS = TIMEOUT_SECONDS * 1000 # Don't modify this
+
+POLLING_TYPES = ['autoTail', 'browse-to-sandbox']
+
+
+class TaskPoller extends Backbone.View
+
+    initialize: ({@requestId, @autoTailFilename, @taskPollTimestamp, @pollingType}) ->
+
+        @history     = new HistoricalTasks [], {params: {requestId: @requestId}}
+
+        @activeTasks = new RequestTasks [],
+            requestId: @requestId
+            state:    'active'
+
+        @activeTasks.fetch().error    @ignore404
+
+        if @history.currentPage is 1
+            @history.fetch().error    @ignore404
+
+        if @pollingType not in POLLING_TYPES
+            throw new Error "#{@pollingType} is not a valid polling type for the task poller. Valid polling types are: #{POLLING_TYPES}"
+
+
+    # Start polling for task changes, and check
+    # Task History changes in case we need
+    # to back out of the file redirect
+    startTaskPolling: ->
+        @showTaskPollWaitingDialog()
+        @stopTaskPolling()
+
+        @listenTo @history, 'reset', @handleHistoryReset
```

Comment:
Now that we are assigning a runId to everything, there is in fact an endpoint to get a task by runId. We can probably use that to be a bit more exact/not get data we aren't going to use